filecoin/activeretrievals.go

```diff
@@ -132,6 +132,22 @@ func (arm *ActiveRetrievalsManager) GetStatusFor(retrievalCid cid.Cid, phase rep
 	return ar.retrievalId, ar.retrievalCid, phaseStart, true
 }
 
+// GetActiveRetrievalCountFor returns the number of active retrievals for a
+// storage provider. Active retrievals only count the retrieval phase, not the
+// query phase requests.
+func (arm *ActiveRetrievalsManager) GetActiveRetrievalCountFor(storageProviderId peer.ID) uint {
+	arm.lk.RLock()
+	defer arm.lk.RUnlock()
+
+	var currentRetrievals uint
```

Comment:
FYC: maybe an unlocked private method here to dedup logic in SetRetrievalCandidate?